includes/admin/class-amp-admin-pointer.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Admin pointer class.
+ *
+ * @package AMP
+ * @since 1.0
+ */
+
+/**
+ * AMP_Admin_Pointer class.
+ *
+ * Outputs an admin pointer to show the new features of v1.0.
+ * Based on https://code.tutsplus.com/articles/integrating-with-wordpress-ui-admin-pointers--wp-26853
+ *
+ * @since 1.0
+ */
+class AMP_Admin_Pointer {
+
+	/**
+	 * The ID of the template mode admin pointer.
+	 *
+	 * @var string
+	 */
+	const TEMPLATE_POINTER_ID = 'amp_template_mode_pointer_10';
+
+	/**
+	 * The slug of the script.
+	 *
+	 * @var string
+	 */
+	const SCRIPT_SLUG = 'amp-admin-pointer';
+
+	/**
+	 * Initializes the class.
+	 *
+	 * @since 1.0
+	 */
+	public function init() {
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_pointer' ) );
+	}
+
+	/**
+	 * Enqueues the pointer assets.
+	 *
+	 * If the pointer has not been dismissed, enqueues the style and script.
+	 * And outputs the pointer data for the script.
+	 *
+	 * @since 1.0
+	 */
+	public function enqueue_pointer() {
+		$dismissed = explode( ',', strval( get_user_meta( get_current_user_id(), 'dismissed_wp_pointers', true ) ) );
```

Comment:
To optimize this code, I've move the `get_user_meta()` out of it.  Then check if there are any dismissed pointers to consider. If yes, explode then. 

To keep the method focused on the enqueueing tasks, I'd abstract it into a new method:

```
	protected function has_pointer_been_dismissed() {
		$dismissed = get_user_meta( get_current_user_id(), 'dismissed_wp_pointers', true );
		if ( empty( $dismissed ) ) {
			return false;
		}
		$dismissed = explode( ',', strval( $dismissed ) );

		return in_array( self::TEMPLATE_POINTER_ID, $dismissed, true );
	}
```

Reply:
Correct. If the user has never dismissed a pointer, then the user meta does not exist or is empty.  In that case, it optimizes the code in that it does not to run `strval()`,  `explode()`, or `in_array()`.

Reply:
I also find it expresses the intent more clearly to help us more quickly read and understand what's going on.

Reply:
@kienstra You could name that guard clause method something less passive such as 

`is_pointer_dismissed()`

Reply:
Thanks, commit [e60f0](https://github.com/Automattic/amp-wp/pull/1271/commits/e60f0f67fced870d11e9e87690198cc32ad39db8) adds `is_pointer_dismissed()`.